gateway-server/src/main/java/org/apache/knox/gateway/services/security/impl/JettySSLService.java

```diff
@@ -236,6 +236,17 @@ public Object buildSslContextFactory(GatewayConfig config) throws AliasServiceEx
     return sslContextFactory;
   }
 
+  @Override
+  public void excludeTopologyFromClientAuth(SslContextFactory sslContextFactory, GatewayConfig config, String topologyName) {
+    if(config.getClientAuthExclude() != null && topologyName != null
+            && config.isClientAuthNeeded() && config.getClientAuthExclude().contains(topologyName)) {
```

Comment:
Most of these conditions could be implemented in GatewayConfigImpl to be self-contained. For instance:
`config.isClientAuthExclude(topologyName)` could return `true` if the desired configs are set and the topology is meant to be excluded.